app/assets/javascripts/modules/highlight-active-section-heading.js

```diff
@@ -1,148 +1,166 @@
-;(function (Modules, root) {
-  'use strict'
-
-  var $ = root.$
-  var $window = $(root)
-
-  Modules.HighlightActiveSectionHeading = function () {
-    var self = this
-    var _hasResized = true
-    var _hasScrolled = true
-    var _interval = 50
-    var anchorIDs = []
-
-    self.getWindowDimensions = function () {
-      return {
-        height: $window.height(),
-        width: $window.width()
-      }
-    }
-
-    self.getWindowPositions = function () {
-      return {
-        scrollTop: $window.scrollTop()
-      }
-    }
-
-    self.getElementOffset = function ($el) {
-      return $el.offset()
-    }
+window.GOVUK = window.GOVUK || {}
+window.GOVUK.Modules = window.GOVUK.Modules || {};
+
+(function (Modules) {
+  function HighlightActiveSectionHeading ($module) {
+    this.$module = $module
+    this._hasResized = true
+    this._hasScrolled = true
+    this._interval = 50
+    this.anchorIDs = []
+  }
 
-    self.start = function ($el) {
-      $window.resize(self.hasResized)
-      $window.scroll(self.hasScrolled)
+  HighlightActiveSectionHeading.prototype.init = function () {
+    window.addEventListener('resize', this.hasResized.bind(this))
+    window.addEventListener('scroll', this.hasScrolled.bind(this))
 
-      setInterval(self.checkResize, _interval)
-      setInterval(self.checkScroll, _interval)
+    setInterval(this.checkResize.bind(this), this._interval)
+    setInterval(this.checkScroll.bind(this), this._interval)
 
-      self.$anchors = $el.find('.js-page-contents a')
-      self.getAnchors()
+    this.anchors = this.$module.querySelectorAll('.js-page-contents a')
+    this.getAnchors()
 
-      self.checkResize()
-      self.checkScroll()
-    }
+    this.checkResize()
+    this.checkScroll()
+  }
 
-    self.hasResized = function () {
-      _hasResized = true
-      return _hasResized
-    }
+  HighlightActiveSectionHeading.prototype.hasResized = function () {
+    this._hasResized = true
+    return this._hasResized
+  }
 
-    self.hasScrolled = function () {
-      _hasScrolled = true
-      return _hasScrolled
-    }
+  HighlightActiveSectionHeading.prototype.hasScrolled = function () {
+    this._hasScrolled = true
+    return this._hasScrolled
+  }
```

Comment:
I don't think there's any need for these functions to return - I assume the past one with jQuery was returning the boolean to not stop event handling?

To be honest I'd imagine you could relatively easily inline these with:

```
window.addEventListener('resize', function () { this._hasResized = true }.bind(this))
window.addEventListener('scroll', function () { this._hasScrolled = true }.bind(this))
```

But I understand I could be missing some testing quirk or something that'd make that hard.

Reply:
Nice optimisation, thanks for spotting, have updated.